projects/clr/rocclr/os/os.hpp

```diff
@@ -526,6 +526,62 @@ inline uint Os::ThreadAffinityMask::getNextSet(uint cpu) const {
 
 #endif
 
+/* Mini numa interface instead of numa lib apis */
+namespace numa {
+
+static constexpr uint32_t kBitsPerUInt64 = 8 * sizeof(uint64_t);
+
+/*! \brief Manage Numa policy.
+ *
+ *  \note Works in Linux only, dummy in Windows.
+ */
+class NumaPolicy final {
+public:
+  enum class Policy {
+    kDefault = 0,
+    kPrefered = 1,
+    kBind = 2,
+    kInterleave = 3,
+    kLocal = 4,
+    kPreferedMany = 5,
+    kWeightedInterleave = 6,
+    kMax = 7
+  };
+  NumaPolicy(const uint32_t numa_node_count);
+
+  //! Query memory policy and node bitmask from Linux kernel
+  bool GetMemPolicy();
+
+  //! Check whether node_index is in bitmask for kPrefered and kBind modes
+  bool IsPolicySetAt(const uint32_t node_index);
```

Comment:
same here

Reply:
Indicate node_index won't be changed in function.